06-building-a-program-start-to-finish/writing-tests/output/spec/InvertedIndex.spec.js

```diff
@@ -0,0 +1,211 @@
+        .toEqual(expectedResult);
+    });
+    it('should return false for incorrect document structure', () => {
+      const term = { t1: 'Welcome home', text: 'This is really home' };
+      expect(indexInstance.createIndex(term, 'term')).toBeFalsy();
+    });
+    it('should return false for file with no content', () => {
+      const term = {};
+      expect(indexInstance.createIndex(term, 'term')).toBeFalsy();
+    });
+  });
+  describe('Search Index', () => {
+    it('should search through single files that are indexed', () => {
+      const requiredOutput = { alice: [0],
+        and: [0, 1, 2],
+        unusual: [1, 2],
+        imagination: [0] };
+      const searchTerm = indexInstance
+        .searchIndex('Alice, and her unusual imagination', 'books');
+      expect(Object.keys(searchTerm[0].indexes))
+        .toEqual(Object.keys(requiredOutput));
+      expect(searchTerm[0].indexes).toEqual(requiredOutput);
+    });
+    it('should return false for an empty String', () => {
+      const term = '';
+      expect(indexInstance.searchIndex(term, 'books'))
+      .toBeFalsy();
+    });
+    it('should return an empty object for an words not found', () => {
+      const term = 'Aeroplane';
+      const expectedOutput = indexInstance.searchIndex(term, 'books');
+      expect(expectedOutput[0].indexes).toEqual({ });
+    });
+  });
+  describe('Tokenize words', () => {
+    it('should strip out special characters from excerpt in documents', () => {
+      let excerpt = 'Alice l##$oves her ima&&gination?';
+      const expectedTokens = ['alice', 'loves', 'her', 'imagination'];
+      excerpt = InvertedIndex.tokenize(excerpt);
+      expect(excerpt).toEqual(expectedTokens);
+    });
+  });
+
+  describe('Get index', () => {
+    it('should return false for an empty filename', () => {
+      const filename = 'books';
+      const expectedOutput = { alice: [0],
+        falls: [0],
+        into: [0],
+        a: [0, 1, 2],
+        rabbit: [0],
+        hole: [0],
+        and: [0, 1, 2],
+        enters: [0],
+        world: [0],
+        full: [0],
+        of: [0, 1, 2],
+        imagination: [0],
+        in: [0],
+        wonderland: [0],
+        an: [1, 2],
+        unusual: [1, 2],
+        alliance: [1, 2],
+        man: [1, 2],
+        elf: [1, 2],
+        dwarf: [1, 2],
+        wizard: [1, 2],
+        hobbit: [1, 2],
+        seek: [1, 2],
+        to: [1, 2],
+        destroy: [1, 2],
+        powerful: [1, 2],
+        ring: [1, 2],
+        the: [1, 2],
+        lord: [1, 2],
+        rings: [1, 2],
+        fellowship: [1, 2] };
+      indexInstance.createIndex(books, 'books');
+      expect(indexInstance.getIndex(filename))
+        .toEqual(expectedOutput);
+    });
+    it('should return the appropriate output for the given filename', () => {
+      const filename = '';
+      indexInstance.createIndex(books, 'books');
+      expect(indexInstance.getIndex(filename))
+        .toBeFalsy();
+    });
+  });
```

Comment:
Please review your test specification for your getIndex method. I believe you have them mixed up.